torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -447,8 +447,8 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
     // Record collective sizes for debug. We only record the size on the first
     // device as multi-device per process is deprecated
-    size_t numelIn_ = -1;
-    size_t numelOut_ = -1;
```

Comment:
This change is ok to me but I forgot why we originally want to make it -1.... I checked the original PR https://github.com/pytorch/pytorch/pull/100413 this should be fine. Thanks for the fix.